libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -117,27 +117,29 @@ void DataFlowAnalyzer::operator()(ForLoop& _for)
 	for (auto& statement: _for.pre.statements)
 		visit(statement);
 
+	boost::optional<Assignments> outerAssignmentsSinceCont = boost::none;
+	swap(outerAssignmentsSinceCont, m_assignmentsSinceCont);
+
 	Assignments assignments;
 	assignments(_for.body);
 	assignments(_for.post);
 	clearValues(assignments.names());
 
 	visit(*_for.condition);
 	(*this)(_for.body);
+	if (m_assignmentsSinceCont)
+		clearValues(m_assignmentsSinceCont->names());
```

Comment:
Won't this be always empty?

Reply:
Since `m_assignmentsSinceCont` isn't called on `Assignment`s

Reply:
It's being updated in `operator()(Continue&)`'s body.

Reply:
Yes, but why isn't it still empty? Shouldn't it call `m_assignmentsSinceCont(_assignment)` for `Assignment`?

Reply:
Ah, now I believe I got your point! Of course, but then I've to rewrite the code quite a bit. Hold on. :)